sources/engine/Stride.Engine/Engine/InstancingBase.cs

```diff
@@ -0,0 +1,31 @@
+using Stride.Core;
+using Stride.Core.Mathematics;
+using Stride.Engine.Processors;
+
+namespace Stride.Engine
+{
+    [DataContract]
+    public abstract class InstancingBase : IInstancing
```

Comment:
I wonder if we want this class to implement `IInstancing`: it forces all `IInstancing` properties to have a setter (because we can't add one in inheriting class). However, those properties might not be something we want visible on user class.
For example, `InstancingUserArray.InstanceWorldBuffer` could be getter only (with internal set) or even totally hidden since it's not supposed to be visible to user.
Same goes for `InstanceCount` which might be `get` only or `get/set` depending on which implementation if `IInstancing` we are talking about.

Reply:
indeed, this can be modeled a bit clearer, I'll look into it.

Reply:
OK, removed `InstancingBase` altogether with latest commit.
Let me know if everything still work as intended, thanks!